### PR TITLE
# Resolve fvt problems

### PR DESCRIPTION
**Pull request by immmus**
@planetf1 Hi, having same read a documentation of junit 5 I found how disable parallel and enable order. Now all test works and passed.

https://gist.github.com/immmus/585ba8fd9ff6ab9013d695e3d0b5c8c5

### CODE REVIEW

File: open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/test/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/junit/AllIT.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.odpi.openmetadata.accessservices.subjectarea.fvt.*;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+@Execution(ExecutionMode.SAME_THREAD)
```
Comment (planetf1) [inline open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/test/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/junit/AllIT.java:16]: Thanks - I missed this when looking. Though I know @davidradl is looking at fixing the underlying issue, so will let him decide whether you want to go with this fix or not 

It's good to know how to get them working, so thanks for sharing the annotation.

I think the other aspect is which way @davidradl thinks makes sense to structure the subject area FVTs in future. I was envisioning that perhaps you would gradually move some of the tests into the JUnit classes themselves, which is why I created one test class in parallel with the FVT classes. Of course you can't order that way so understand why you suggest the change (I tried the same but missed that extra annotation!)

It probably depends whether the other fixes are near term or not, but I think we'd all agree we don't want to depend on single thread execution and ordering to allow them to work in future

Will leave to you both to agree. I'm ok with either . See #3312 #3304 #3302  which all seem to have some overlap

File: open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/test/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/junit/AllIT.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
```
Comment (planetf1) [inline open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/test/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/junit/AllIT.java:17]: and this annotation. I misread as being default. :-)

**Comment by immmus** [comment]
@planetf1 I think that this solution will look good as temporary, until all subsequent changes are made. Change no longer make any labor. We will get all work test during maven build now.

File: open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/test/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/junit/AllIT.java (hunk 2 above)
Comment (immmus) [reply open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/test/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/junit/AllIT.java:17]: I initially thought so too. It is strange that this is not so)

**Comment by planetf1** [comment]
@immmus @davidradl I think we should close this PR now. No need to merge since #3304 has fixed the underlying issues, so no need to enforce ordering. It was good to learn this tip though for the future so thanks

**Comment by immmus** [comment]
Yes, I close this.

**Pull request closed without merging**